Extensions/TkSharp.Extensions.LibHac/Extensions/FileSystemExtensions.cs

```diff
@@ -25,7 +32,13 @@ public static SwitchFs GetSwitchFs(this IStorage storage, string filePath, KeySe
             _ => throw new ArgumentException($"Unsupported file extension: '{extension}'", nameof(filePath)),
         };
     }
-
+    private static bool IsXci(IStorage storage)
+    {
+        Span<byte> buffer = stackalloc byte[4];
+        storage.Read(0x100, buffer).ThrowIfFailure();
+        return buffer.SequenceEqual(stackalloc byte[] { (byte)'H', (byte)'E', (byte)'A', (byte)'D' });
```

Comment:
This can be simplified.

```suggestion
        return buffer.SequenceEqual("HEAD"u8);
```